Basic HTML, CSS, & JS outlines with sample content

daniellehu opened the pull request:
Allows for a basic outline of Wiki-Gen. Later contributions would involve extracting text from the database and systematically formatting the text into this format.

For this outline, refer to https://en.wikipedia.org/wiki/Eurasian_sparrowhawk to compare the two websites. Format should be closely related.

Unnecessary photos (i.e. 250px-Wikipedia-logo-v2-en.svg.png and  photos/wikigenIcon_48x48.png from the photos folder) could be removed before merging to the main branch since they were not directly used for the outline.

Commit:
Basic HTML, CSS, & JS outlines with sample content

kevinzhe commented:
Looks great so far! I have some minor refactors I'd like to make just for code uniformity. Other than that, I have a few suggestions before pulling this into the master branch:
- Search bar's vertical sizing changes with window width in the mobile breakpoint, but I think it should be a constant height
- Change 'search' to 'generate'?
- Make the breakpoint to the mobile view a bit lower (bootstrap sm rather than md)
- Since bootstrap glyphicons are PNGs (ie. do not scale), let's use [FontAwesome](http://fortawesome.github.io/Font-Awesome/) instead

kevinzhe committed:
Refactor and organize code

Organize HTML/CSS/JS into respective directories. Add local
Bootstrap/jQuery. Change variable names from camelCase to
dashed. Add a .gitignore.

Commit:
Several small improvements made

Based on pull request discussion, the following changes were made:
1. HTML & CSS code style change from camelCase to dashes for unity
purposes; JS code kept in camelCase since most of jQuery uses camelCase
and JS doesn't accept hyphens in variable names.
2. Added internal references to JS and Bootstrap, rather than
referencing to a CDN.
3. Separated files into respective directories
4. Added .gitignore file
5. Changed breakpoint to mobile view smaller (from 992px to 768px width)
6. Changed "Search" to "Generate"
7. Changed navigation icon (menu-hamburger) and search icon to
font-awesome for scalability purposes (but kept arrow glyphicons from
Bootstrap since font-awesome did not have any icons resembling the ones
found on Wikipedia's mobile site, and the need for scalable arrows on a
mobile platform seems unneccessary)

Commit:
fixed collision issues

Commit:
fixed collision issues

daniellehu commented:
Note: I made some mistakes when making changes which led to some collisions, but everything should be fixed now! If any issues pop up, let me know.

Take a look at the new files/website; you should see your suggestions in the new website:
- Search bar's vertical sizing changed to constant height
- Changed "search" to "generate" (good call!)
- Mobile breakpoint now at 768px rather than 992px
- FontAwesome used for search icon and menu-hamburger icon, but I kept the glyphicon up and down arrows for the mobile contents section since there were no similar icons with FontAwesome (and scalability shouldn't be a major issue in mobile)

Additionally:
- I fixed a minor issue when going from the mobile form to desktop form where the content's paragraph sections remained hidden. Also, the issue of being able to click to minimize/expand these content paragraph sections in desktop mode was fixed.
- Used .gitignore to remove the unneeded images and Adobe Illustrator file

If there are any more changes to be done before merging to master, let me know!

kevinzhe commented:
Looks great! We can handle some of the final polishing later. I've merged this branch into `topic/flask-init`, so we can begin to incorporate this static page as a dynamic template.

I'll close this PR for now, we can continue discussion in another PR over in `topic/flask-init` when time comes.
